mmpose/datasets/datasets/top_down/topdown_mpii_dataset.py

```diff
@@ -0,0 +1,218 @@
+            })
+
+        return gt_db
+
+    def evaluate(self, outputs, res_folder, metrics='PCKh', **kwargs):
+        """Evaluate PCKh for MPII dataset. Adapted from
+        https://github.com/leoxiaobin/deep-high-resolution- net.pytorch
+        Original licence: Copyright (c) Microsoft, under the MIT License.
+
+        Note:
+            num_keypoints: K
+
+        Args:
+            outputs(list(preds, boxes, image_path)):Output results.
+                preds(np.ndarray[1,K,3]): The first two dimensions are
+                    coordinates, score is the third dimension of the array.
+                boxes(np.ndarray[1,6]): [center[0], center[1], scale[0]
+                    , scale[1],area, score]
+                image_path(list[str]): For example, ['0', '0',
+                    '0', '0', '0', '1', '1', '6', '3', '.', 'j', 'p', 'g']
+            res_folder(str): Path of directory to save the results.
+            metrics(str): Metrics to be performed.
+                Defaults: 'PCKh'.
+
+        Returns:
+            PCKh for each joint
+        """
+
+        # only PCKh is supported.
+        assert metrics == 'PCKh'
+
+        preds = np.stack([kpts[0] for kpts, _, _ in outputs])
+
+        # convert 0-based index to 1-based index
+        preds[..., :2] += 1.0
+
+        if res_folder:
+            pred_file = os.path.join(res_folder, 'pred.mat')
+            savemat(pred_file, mdict={'preds': preds})
+
+        SC_BIAS = 0.6
+        threshold = 0.5
+
+        gt_file = os.path.join(
+            os.path.dirname(self.ann_file), 'mpii_gt_valid.mat')
+        gt_dict = loadmat(gt_file)
+        dataset_joints = gt_dict['dataset_joints']
+        jnt_missing = gt_dict['jnt_missing']
+        pos_gt_src = gt_dict['pos_gt_src']
+        headboxes_src = gt_dict['headboxes_src']
+
+        pos_pred_src = np.transpose(preds, [1, 2, 0])
+
+        head = np.where(dataset_joints == 'head')[1][0]
+        lsho = np.where(dataset_joints == 'lsho')[1][0]
+        lelb = np.where(dataset_joints == 'lelb')[1][0]
+        lwri = np.where(dataset_joints == 'lwri')[1][0]
+        lhip = np.where(dataset_joints == 'lhip')[1][0]
+        lkne = np.where(dataset_joints == 'lkne')[1][0]
+        lank = np.where(dataset_joints == 'lank')[1][0]
+
+        rsho = np.where(dataset_joints == 'rsho')[1][0]
+        relb = np.where(dataset_joints == 'relb')[1][0]
+        rwri = np.where(dataset_joints == 'rwri')[1][0]
+        rkne = np.where(dataset_joints == 'rkne')[1][0]
+        rank = np.where(dataset_joints == 'rank')[1][0]
+        rhip = np.where(dataset_joints == 'rhip')[1][0]
+
+        jnt_visible = 1 - jnt_missing
+        uv_error = pos_pred_src - pos_gt_src
+        uv_err = np.linalg.norm(uv_error, axis=1)
+        headsizes = headboxes_src[1, :, :] - headboxes_src[0, :, :]
+        headsizes = np.linalg.norm(headsizes, axis=0)
+        headsizes *= SC_BIAS
+        scale = headsizes * np.ones((len(uv_err), 1), dtype=np.float32)
```

Comment:
is `np.full` useful here? https://numpy.org/doc/stable/reference/generated/numpy.full.html

Reply:
what's the shape of headsizes, and what's the purpose of this multiplication?

Reply:
The shape of `headsizes` is (2958, )   and `len(uv_err)=16`.
And the multiplication will get a np.array shaped (16, 2958).